tensorflow/core/kernels/mkl_maxpooling_op.cc

```diff
@@ -83,6 +85,17 @@ class MklMaxPoolingOp : public MklPoolingForwardOpBase<T> {
         const int kOutputIndex = 0;
         this->AllocateEmptyOutputTensor(context, kOutputIndex, &pool_params,
                                         output_dims_mkl_order, &output_tensor);
+        bool int8_forward_inference =
+            std::is_same<T, qint8>::value || std::is_same<T, quint8>::value;
+
+        if (!int8_forward_inference) {
```

Comment:
Just curious, why don't we need this for `int8` too?

Reply:
Int8/Quantized version of MKL MaxPooling Op does not have that extra workspace tensor defined. For some reason, this workspace tensor is unique to MaxPooling FP32 op. @yiqianglee can help clarify please.

Reply:
Workspace as second output tensor only happen on MklMaxPooling's OP definition, in MklQuantizeMaxPool, it doesn't need workspace tensor (because workspace tensor is used by MaxPool's backward op, int8 is inference only).
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/ops/nn_ops.cc#L2168
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/ops/mkl_nn_ops.cc#L120

Reply:
Thank you for the explanation!